lib/Dialect/FIRRTL/FIRRTLTypes.cpp

```diff
@@ -909,13 +987,19 @@ bool firrtl::areTypesConstCastable(FIRRTLType destFType, FIRRTLType srcFType,
   if (destFType == srcFType)
     return true;
 
-  auto destType = destFType.dyn_cast<FIRRTLBaseType>();
-  auto srcType = srcFType.dyn_cast<FIRRTLBaseType>();
+  auto destType =
```

Comment:
Either this should/can be `firrtl::type_cast` or we can't blindly call `getAnonymousType()` on it as it may be NULL.